src/providers/oracle/qgsoracleconn.cpp

```diff
@@ -632,39 +632,98 @@ void QgsOracleConn::retrieveLayerTypes( QgsOracleLayerProperty &layerProperty, b
   }
 }
 
-QString QgsOracleConn::databaseTypeFilter( const QString &alias, QString geomCol, QgsWkbTypes::Type geomType )
+QString QgsOracleConn::databaseTypeFilter( const QString &alias, QString geomCol, QgsWkbTypes::Type geomType, QString pk, QString tableName )
 {
   geomCol = quotedIdentifier( alias ) + "." + quotedIdentifier( geomCol );
 
+  // Oracle DB allows tables to contain several geometry types (including also single and multi geometry).
+  // This behavior is not wished in QGIS. Indeed, as other provider don't handle this feature, QgsVectorLayer is supposed to allow only one geometry type.
+  // In Oracle DB, geometry type are determined by sdo_gtype and sdo_interpretation (cf. https://docs.oracle.com/database/121/SPATL/sdo_geometry-object-type.htm)
+  // sdo_interpretation is a part of sdo_elem_info, which can be extracted like explained in the following discussion :
+  // https://stackoverflow.com/questions/3971693/sql-query-to-determine-if-oracle-spatial-table-contains-curves
   switch ( geomType )
   {
     case QgsWkbTypes::Point:
     case QgsWkbTypes::Point25D:
+      return QStringLiteral( " mod(%1.sdo_gtype,100) = 1" ).arg( geomCol );
     case QgsWkbTypes::MultiPoint:
     case QgsWkbTypes::MultiPoint25D:
-      return QStringLiteral( "mod(%1.sdo_gtype,100) IN (1,5)" ).arg( geomCol );
+      return QStringLiteral( " mod(%1.sdo_gtype,100) = 5" ).arg( geomCol );
     case QgsWkbTypes::LineString:
     case QgsWkbTypes::LineString25D:
     case QgsWkbTypes::LineStringZ:
+      // LineString gtype ends with 2, sdo_interpretation must not be equal to 2 (CircularString) or 3 (Nurbs)
```

Comment:
I think this filter will decrease performances, because it requests all interpretation from the table and then join on PK. It is worth trying on big dataset but I'm pretty sure this one should be better   

```sql
SELECT "pk" FROM QGIS.LINE_DATA l
WHERE mod(l."GEOM"."SDO_GTYPE",100) = 2
AND (
    SELECT MAX(DECODE(MOD(ROWNUM, 3), 0, t.COLUMN_VALUE, NULL)) 
    FROM TABLE(l."GEOM"."SDO_ELEM_INFO") t
    ) not in (2,3);
```

We should also put this request in a QString and reuse it in all following filters with just parameter for *sdo_gtype* and *in/not in* 2,3/4